src/libs/tracking/utils.ts

```diff
@@ -0,0 +1,164 @@
+// @see https://raw.githubusercontent.com/PostHog/posthog-js/master/src/utils.js
+
+import BrowserInfo = browser.runtime.BrowserInfo;
+
+type Browser =
+  | 'Opera Mini'
+  | 'Opera'
+  | 'BlackBerry'
+  | 'Internet Explorer Mobile'
+  | 'Samsung Internet'
+  | 'Microsoft Edge'
+  | 'Facebook Mobile'
+  | 'Chrome'
+  | 'Chrome iOS'
+  | 'UC Browser'
+  | 'Firefox iOS'
+  | 'Mobile Safari'
+  | 'Safari'
+  | 'Android Mobile'
+  | 'Konqueror'
+  | 'Firefox'
+  | 'Internet Explorer'
+  | 'Mozilla'
+  | '';
+
+/**
+ * This function detects which browser is running this script.
+ * The order of the checks are important since many user agents
+ * include key words used in later checks.
+ */
+const getBrowser = (userAgent: string, vendor: string): Browser => {
+  vendor = vendor || ''; // vendor is undefined for at least IE9
+  if (userAgent.includes(' OPR/')) {
+    if (userAgent.includes('Mini')) {
+      return 'Opera Mini';
+    }
+    return 'Opera';
+  } else if (/(BlackBerry|PlayBook|BB10)/i.test(userAgent)) {
+    return 'BlackBerry';
+  } else if (
+    userAgent.includes('IEMobile') ||
+    userAgent.includes('WPDesktop')
+  ) {
+    return 'Internet Explorer Mobile';
+  } else if (userAgent.includes('SamsungBrowser/')) {
+    // https://developer.samsung.com/internet/user-agent-string-format
+    return 'Samsung Internet';
+  } else if (userAgent.includes('Edge') || userAgent.includes('Edg/')) {
+    return 'Microsoft Edge';
+  } else if (userAgent.includes('FBIOS')) {
+    return 'Facebook Mobile';
+  } else if (userAgent.includes('Chrome')) {
+    return 'Chrome';
+  } else if (userAgent.includes('CriOS')) {
+    return 'Chrome iOS';
+  } else if (userAgent.includes('UCWEB') || userAgent.includes('UCBrowser')) {
+    return 'UC Browser';
+  } else if (userAgent.includes('FxiOS')) {
+    return 'Firefox iOS';
+  } else if (vendor.includes('Apple')) {
+    if (userAgent.includes('Mobile')) {
+      return 'Mobile Safari';
+    }
+    return 'Safari';
+  } else if (userAgent.includes('Android')) {
+    return 'Android Mobile';
+  } else if (userAgent.includes('Konqueror')) {
+    return 'Konqueror';
+  } else if (userAgent.includes('Firefox')) {
+    return 'Firefox';
+  } else if (userAgent.includes('MSIE') || userAgent.includes('Trident/')) {
+    return 'Internet Explorer';
+  } else if (userAgent.includes('Gecko')) {
+    return 'Mozilla';
+  } else {
+    return '';
+  }
+};
+
+/**
+ * This function detects which browser version is running this script,
+ * parsing major and minor version (e.g., 42.1). User agent strings from:
+ * http://www.useragentstring.com/pages/useragentstring.php
+ */
+const getBrowserVersion = (userAgent: string, vendor: string) => {
```

Comment:
you don't want use a module like: https://www.npmjs.com/package/useragent?
It's could be easier to maintain

Reply:
I wish but... I'm using what's __PostHog__ is using, to be coherent with the values __PostHog__ is using...
And this method is only used as a polyfill for Chrome since Firefox web extension runtime comes with a function doing juste dat already: 
https://developer.mozilla.org/fr/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getBrowserInfo

> also about the library we might use only 1% of its features